### ISSUE TITLE
chapter 4 noise diagrams

### ISSUE THREAD

**Issue by leeprinxin** [GANs-in-Action/gans-in-action#17]
I use the colab environment to run it.
keras or tensorflow.keras: 2.4.3 or 2.4.0.
tensorflow: 2.4.1

The code is the same, but only noise diagrams appear after running.
I tried replacing the optimiser with RMSprop and it also only produces noise diagrams

**Comment by leeprinxin** [comment]
code link: https://colab.research.google.com/gist/leeprinxin/967ce5c24b163c68d13ec5305dea7207/dcgan.ipynb

**Comment by g20tk009** [comment]
me too

**Comment by g20tk009** [comment]
I add two **Dropout** into discriminator and I get the no noisy picture but I don't know it is expected or not.
I would like to know what should I do.

```python
from keras.layers import Flatten, Dropout
from keras.layers.convolutional import Conv2D

def build_discriminator(img_shape):

  model = Sequential()

  # model.add(Flatten(input_shape=img_shape))
  model.add(Conv2D(32, kernel_size=3, strides=2, input_shape=img_shape, padding='same'))
  model.add(LeakyReLU(alpha=0.01))

  # model.add(Dense(128))
  # model.add(LeakyReLU(alpha=0.01))
  model.add(Conv2D(64, kernel_size=3, strides=2, padding='same'))
  # model.add(BatchNormalization())
  model.add(LeakyReLU(alpha=0.01))
  model.add(Dropout(0.5))

  model.add(Conv2D(128, kernel_size=3, strides=2, padding='same'))
  # model.add(BatchNormalization())
  model.add(LeakyReLU(alpha=0.01))
  model.add(Dropout(0.5))
  
  # model.add(Dense(1, activation='sigmoid'))
  model.add(Flatten())
  model.add(Dense(1, activation='sigmoid'))

  return model
```

**Comment by g20tk009** [comment]
https://twitter.com/Satoru46786131/status/1384487141560909824?s=20
